task14/index.js

```diff
@@ -0,0 +1,109 @@
+const parent = document.getElementById("js-parent"); 
+const modalPlace = document.getElementById('js-modalPlace');
+
+const appendModalForParent = () => {
+    const modalText = document.createElement('p');
+    const getNumber = document.createElement('input')
+    const closeBtn = document.createElement('input');
+    modalPlace.className = "modal";
+    modalPlace.style.display = "block";
+    modalText.textContent = "数字を入力して値を取得します。";
+    getNumber.id = "js-number";
+    getNumber.className = "number";
+    getNumber.type = "number";
+    closeBtn.id = "js-closeBtn";
+    closeBtn.className = "close-button";
+    closeBtn.type = "button";
+    closeBtn.value = "click";
+
+    modalPlace.appendChild(modalText);
+    modalPlace.appendChild(getNumber);
+    modalPlace.appendChild(closeBtn);
+};
+
+const appendButtonForParent = () => {
+    const button = document.createElement('input');
+    button.id = "js-openBtn";
+    button.type = "button";
+    button.value = "click";
+    parent.appendChild(button);
+};
+appendButtonForParent();
+
+const requestNumber = () => {
+    const getValue = document.getElementById("js-number");
+    if(getValue.value === ""){
+        alert("数字を入力してください");
+    } else {
+        console.log(getValue.value);
+        modalPlace.style.display = "none";
+        init();
+    }
+};
+
+const openBtn = document.getElementById("js-openBtn");
+openBtn.addEventListener("click", (e) => {
+    appendModalForParent();
+    e.target.remove();
+    modalClose();
+});
+
+const modalClose = () => {
+    const closeBtn = document.getElementById("js-closeBtn");
+    closeBtn.addEventListener("click", () => {
+        requestNumber();
+    });
+};
+
+const getFetchData = async () => {
+    const response = await fetch("https://jsondata.okiba.me/v1/json/3jhPW210812052923");
+    const json = await response.json();
+    return json.data;
+};
+
+const addLoading = () => {
+    const loadingPlace = document.createElement("div");
+    const gif = document.createElement('img');
+    loadingPlace.id = "loadingPlace";
+    gif.src = "loading-circle.gif";
+    parent.appendChild(loadingPlace).appendChild(gif);
+};
+
+const removeLoading = () => {
+    document.getElementById("loadingPlace").remove();
+};
+
+const appendListForParent = (value) => {
+    const ul = document.createElement('ul');
+    const fragment = document.createDocumentFragment();
+    
+    value.forEach(d => {
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+    
+        a.textContent = d.text;
+        a.href = d.a;
+        img.src = d.img;
+        img.alt = d.alt;
```

Comment:
[Q]
1. Does `d` stand for `data`?
2. Why did you choose `d`?

Reply:
Thank you for your comment.
I'm sorry for the delay in responding to your review.

1. That's right! It stands for data.
2.When I asked my teacher about writing with "d", he told me that data is the same as data in both singular and plural systems, so it can be abbreviated to "d" in this context. In this context, "d" is a good abbreviation. In this case, I wrote "d" with the meaning that I learned that it is possible to write arguments in this way.